flask_appbuilder/security/manager.py

```diff
@@ -878,9 +878,15 @@ def _search_ldap(self, ldap, con, username):
                 filter_str, request_fields, self.auth_ldap_search
             )
         )
-        search_result = con.search_s(
-            self.auth_ldap_search, ldap.SCOPE_SUBTREE, filter_str, request_fields
-        )
+
+        # filter out any referrals that get returned
+        search_result = [
+            (dn, attrs)
+            for dn, attrs in con.search_s(
+                self.auth_ldap_search, ldap.SCOPE_SUBTREE, filter_str, request_fields
+            )
+            if dn is not None
```

Comment:
I'm pretty sure its a bad idea to just ignore any search reults with None for `dn`.

Are we 100% sure the only case this occurs in is SearchResultReference, because I doubt this.

Isn't there a better way to filter SearchResultReference instances explicitly.

Reply:
There doesn't seem to be, setting the `ldap.OPT_REFERRALS` prevents referrals from being chased, but not from being returned.  I based my thought process off this [email](https://mail.python.org/pipermail/python-ldap/2014q1/003350.html) in the `python-ldap` maiiling list, which recommended this approach I took above, granted this email is rather old.

Reply:
Looks like the python code that wraps the c-extension code in `python-ldap` hides this information away, and there seems to be now way to filter this out in a more explicit manner.  Looking at how the surround code treats a `None` dn and given that we return `(None, None)` when we get an empty search, there is no change in behavior this causes.  We could perhaps split this up and do the filtering on another line if we want to log the original search results.

Reply:
Ok, I've made some adjustments to the original search results are still logged and the filtering is more specific.